File: 1863_leetcode.py
from functools import lru_cache
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        @lru_cache(None)
        def dfs(index, xor):
            if index >= len(nums):
                return xor
            pick = dfs(index + 1, nums[index] ^ xor)  # include current element
            skip = dfs(index + 1, xor)                # exclude current element
            return pick + skip
        return dfs(0, 0)

File: 1863_leetcode.go
func subsetXORSum(nums []int) int {
    n := len(nums)
    subsetCount := 1 << n // 2^n subsets
    totalSum := 0

    // Iterate over all possible subsets
    for subsetMask := 0; subsetMask < subsetCount; subsetMask++ {
        currentXOR := 0
        // Calculate XOR for the current subset
        for i := 0; i < n; i++ {
            if subsetMask&(1<<i) != 0 {
                currentXOR ^= nums[i]
            }
        }
        totalSum += currentXOR
    }

    return totalSum
}